src/main/java/com/gooddata/http/client/GoodDataHttpClient.java

```diff
@@ -200,30 +223,36 @@ private HttpResponse handleResponse(final HttpHost httpHost, final HttpRequest r
      * </ul>
      * @throws GoodDataAuthException error
      */
-    private boolean refreshTt(final HttpHost httpHost) throws IOException {
+    private boolean refreshTt() throws IOException {
         log.debug("Obtaining TT");
-        final HttpGet getTT = new HttpGet(TOKEN_URL);
+
+        final HttpGet request = new HttpGet(TOKEN_URL);
+        request.addHeader(SST_HEADER, sst);
         try {
-            final HttpResponse response = httpClient.execute(httpHost, getTT, context);
+            final HttpResponse response = httpClient.execute(authHost, request, (HttpContext) null);
             final int status = response.getStatusLine().getStatusCode();
             switch (status) {
                 case HttpStatus.SC_OK:
+                    tt = TokenUtils.extractTT(response);
```

Comment:
This is probably wrong -  the response without accept type is HTML without token String (it's returned as a Cookie).
For proper use it's necessary to add Accept:application/json or application/yaml to the request.